### PR TITLE
Add Face/Edge deconstruction

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: nb/projects/mold/hex.md [hunk 1]
```diff
@@ -13,14 +13,8 @@ const assembled = Hexagon(50, 50, 50)
...
 ```JavaScript
-const disassembled = Hexagon(50, 50, 50)
-  .eachEdge(
-    (e, l) =>
-      Box([0, l], [0, 5], [0, -1])
-        .join(ArcX([0, l], [4, 7.0], 3).hasAngle(0 / 4, 2 / 4))
-        .to(e),
-    (f, e) => f.e(1).join(e).to(XY())
-  )
+const disassembled = assembled
```
Comment (BarbourSmith) [inline nb/projects/mold/hex.md:16]: Well this is simpler :)

File: algorithm/cgal/cgal.cc [hunk 2]
```diff
@@ -4712,28 +4804,43 @@ int Fuse(Geometry* geometry) {
...
+        Surface_mesh cutMeshCopy(geometry->mesh(nth));
+        if (!CGAL::Polygon_mesh_processing::corefine_and_compute_union(
+                geometry->mesh(target), cutMeshCopy, geometry->mesh(target),
+                CGAL::parameters::all_default(),
+                CGAL::parameters::all_default(),
+                CGAL::parameters::all_default())) {
+          return STATUS_ZERO_THICKNESS;
+        }
+      } else {
+        // TODO: Optimize out unnecessary conversions.
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal.cc:4830]: How many unnecessary conversions are happening?

File: algorithm/cgal/eachEdge.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { fromCgalGeometry, withCgalGeometry } from './cgalGeometry.js';
+
+import { ErrorZeroThickness } from './error.js';
+
+export const eachEdge = (inputs, emit) => {
+  return withCgalGeometry(inputs, (cgalGeometry, g) => {
+    const status = g.EachEdge(cgalGeometry, emit);
+    switch (status) {
+      case STATUS_ZERO_THICKNESS:
+        throw new ErrorZeroThickness('Zero thickness produced by eachEdge');
```
Comment (BarbourSmith) [inline algorithm/cgal/eachEdge.js:11]: It's nice that these tell where they are coming from

**Review by BarbourSmith — APPROVED** [review]
Looks good to me!

What's the next step towards unfolding?

It looks like the change in reference plane direction messed with a couple examples:

image

image

And then this one also looks different, but I'm not sure what would cause this:

image

File: algorithm/cgal/cgal.cc (hunk 2 above)
Comment (pentacular) [reply algorithm/cgal/cgal.cc:4830]: Each boolean is converting to and from the manifold representation, so potentially quite a lot.

File: nb/projects/mold/hex.md (hunk 1 above)
Comment (pentacular) [reply nb/projects/mold/hex.md:16]: Yeah, getting there.

Comment (BarbourSmith) [reply nb/projects/mold/hex.md:16]: Aren't you supposed to be on vacation? :)

**Pull request merged**